dependencies/wasm/CMakeLists.txt

```diff
@@ -154,12 +154,16 @@ function(add_wasm_halide_test TARGET)
     endif ()
 
     set(script "require('./${TARGET}.js')")
-    if (WEBGPU_NODE_BINDINGS)
-        set(script "\
-            const provider = require('${WEBGPU_NODE_BINDINGS}')\n\
-            const gpu = provider.create([])\n\
-            const navigator = { gpu: gpu }\n
-            ${script}")
+    if (Halide_TARGET MATCHES "webgpu")
+        if (DEFINED ENV{HL_WEBGPU_NODE_BINDINGS})
+            set(script "\
+                const provider = require('$ENV{HL_WEBGPU_NODE_BINDINGS}')\n\
+                const gpu = provider.create([])\n\
+                const navigator = { gpu: gpu }\n
+                ${script}")
+        else ()
+            message(FATAL_ERROR "You must define the env var HL_WEBGPU_NODE_BINDINGS=/path/to/dawn.node when running wasm tests for Halide.")
+        endif ()
     endif ()
 
```

Comment:
I think you should instead write a launcher script (say `tools/launch_webgpu_test.js` that [reads `process.argv`](https://stackoverflow.com/a/4351548/2137996) to determine which script to `require` and reads `HL_WEBGPU_NODE_BINDINGS` from the environment.

A few benefits:

1. We don't need to generate JavaScript code from CMake
2. CMake doesn't need to try to find `dawn.node` or even know it exists.
3. `HL_WEBGPU_NODE_BINDINGS` is purely an environment variable.

Below, you'd write:

```
add_halide_test("${TARGET}"
                GROUPS ${args_GROUPS}
                COMMAND ${NODE_JS_EXECUTABLE} "${Halide_SOURCE_DIR}/tools/launch_webgpu_test.js" "${TARGET}.js")
```

Reply:
Yeah, this one I think I agree with 100%, I'll put this in place shortly.